fastlane/Fastfile

```diff
@@ -416,7 +421,7 @@ platform :ios do
       body: {
         title: "#{ios_get_app_version} Release: Merge changes from #{ios_get_build_version} into `trunk`",
         body: "Merge changes from #{ios_get_app_version} (#{ios_get_build_version}) to `trunk`.\n\n## To test\n\n- Ensure the build is 🟢\n- The code changes here were tested in their own PRs",
-        head: release_branch_name,
+        head: options[:beta_release] ? merge_branch_name : release_branch_name,
```

Comment:
We might always want to use a `merge/release-x.y-into-trunk` branch in this scenario, for code freeze and betas as well as the release finalization. There are chances of running into merge conflicts during the release finalization as well.

It's a straightforward change to make, though, so we might see how the next couple of releases go and then update?

Reply:
@mokagio I just thought that this approach:

1. Don't create issues for the "final merge" (hotfix and release finalization)
2. It removes the need to delete the branch after the process is finished manually

The original issue I'm trying to prevent is merging `trunk` into the release branch. However, if we just did the final build or the hotfix process, if we run into a conflict there's no problem in fixing those and merging the branch, because the process (in theory) is already finished. And we also remove the manual step of "Ensure the release branch is deleted."

Do you think that makes sense?

Reply:
I like that this would remove the need to "ensure the release branch is deleted."

I'm not sure about merging `trunk` into `release/x.y.z`, even if the release process is finished, because I'd always expect release to go into `trunk`, not vice versa.

However, I realize this might be overly pedantic from a process point of view, particularly given we don't keep track of those branches, but use tags instead. Pragmatically, as long as a release tag comes from a release branch that doesn't have `trunk` merged into it, then we should be good.

Keen to hear what other folks from @Automattic/apps-infrastructure think of this.